File: 0202-Happy-Number/0202.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean isHappy(int n) {
        Set<Integer> visit = new HashSet<>();

        while (!visit.contains(n)) {
            visit.add(n);

            n = sumOfSquare(n);

            if (n == 1)
                return true;
        }
        return false;
    }

    public int sumOfSquare(int n) {
        int output = 0;

        while (n > 0) {
            int digit = n % 10;
            output += Math.pow(digit, 2);
            n /= 10;
        }

        return output;
    }
}

File: 0202-Happy-Number/0202.py
class Solution:
    def isHappy(self, n: int) -> bool:
        visit = set()

        while n not in visit:
            visit.add(n)
            n = self.sumOfSquare(n)

            if n == 1:
                return True
        
        return False
    
    def sumOfSquare(self, n: int) -> bool:
        output = 0

        while n > 0:
            digit = n % 10
            output += digit ** 2
            n = n // 10
        
        return output